src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -33,11 +42,66 @@ def cli_asset():
     default=False,
 )
 @_handle_error
-def asset_snapshot(**args):
+def asset_snapshot(root_dir, manifest_out, recursive, **args):
     """
     Creates manifest of files specified root directory.
     """
-    click.echo("snapshot taken")
+    root_dir_basename = os.path.basename(root_dir) + "_"
+
+    if not os.path.isdir(root_dir):
+        misconfigured_directories_msg = f"Specified root directory {root_dir} does not exist. "
+        raise NonValidInputError(misconfigured_directories_msg)
+
+    if manifest_out and not os.path.isdir(manifest_out):
+        misconfigured_directories_msg = (
+            f"Specified destination directory {manifest_out} does not exist. "
+        )
+        raise NonValidInputError(misconfigured_directories_msg)
```

Comment:
I'm following what the `_submit_job_bundle.py` does with inputs, but changing the error `NonValidInputError`, which is a client error. It seems redundant because I only snapshot one file though, so should I follow the style of other functions or make it less redundant ?